Cython/Compiler/ModuleNode.py

```diff
@@ -3255,8 +3255,11 @@ def generate_module_import_setup(self, env, code):
             # out the module path at runtime by rerunning the import lookup
             code.putln("if (!CYTHON_PEP489_MULTI_PHASE_INIT) {")
             package_name, _ = self.full_module_name.rsplit('.', 1)
+            module_name = env.module_name
             if '.' in package_name:
                 parent_name = '"%s"' % (package_name.rsplit('.', 1)[0],)
+                if sys.version_info > (3, 4):
```

Comment:
We normally try to make the Cython output as version independent as possible. Is there a way that this could be handled in C instead?

It looks like `module_name` is only used in the 3.4+ case, so you could just always do it?

Reply:
`module_name` is used in both paths, see line 453 which remained unchanged. I could change the C code to prefix a '.' to the `PyUnicodeObject` in the 3.4+ code path, but that also feels wrong.

Reply:
I changed this to always prepend a '.' if the module is a subpackage, and added C code to trim the '.' off for pypy3 < 3.4